go_test.go

```diff
@@ -75,12 +76,37 @@ func TestGoGetSpecificVersion(t *testing.T) {
 	assert.NoError(t, os.Chdir(wd))
 }
 
+// Test 'go get' with a nested package (a specific directory inside a package) and validate it was cached successfully.
+func TestGoGetNestedPackage(t *testing.T) {
+	goPath, cleanUpFunc := initGoTest(t)
+	defer cleanUpFunc()
+	buildNumber := "1"
+	wd, err := os.Getwd()
+	assert.NoError(t, err)
+	prepareGoProject("project1", "", t, true)
+	artifactoryGoCli := tests.NewJfrogCli(execMain, "jfrog rt", "")
+	//  When running 'go get github.com/golang/mock/mockgen@v1.4.1':
+	//		* "mockgen" is a directory inside "mock" package ("mockgen" doesn't contain "go.mod").
+	//		* go download and save the whole "mock" package in local cache under 'github.com/golang/mock@v1.4.1' -- >
+	//		  "go get" downloads and saves the whole "mock" package in the local cache under 'github.com/golang/mock@v1.4.1'.
```

Comment:
The above description lines are not very clear IMO.  Let's rephrase and elaborate more so that it's super clear for the random user.
Also, the alignment with the "*" in the beginning is a bit strage and not ecstatic IMO.